include/GafferSceneUI/Private/Inspector.h

```diff
@@ -92,7 +92,7 @@ class GAFFERSCENEUI_API Inspector : public IECore::RefCounted, public boost::sig
 		const std::string &name() const;
 
 		/// Called by the UI to inspect the property in the current context.
-		ResultPtr inspect() const;
+		virtual ResultPtr inspect() const;
```

Comment:
I'd like to keep this non-virtual, so that derived classes have no opportunity to subvert the rules about history traversal, and there is no confusion about which methods they should be overriding.

I'm guessing this was done to avoid the overhead of history generation for non-existing attributes in the ParameterInspector? If so, can we achieve the same by adding a `ParameterInspector::history()` override instead?

Reply:
> I'm guessing this was done to avoid the overhead of history generation for non-existing attributes in the ParameterInspector?

That and there was also a test failing in `ParameterInspectorTest` : 

https://github.com/ericmehl/gaffer/blob/6f6467d549e410040dddce5966cdf2bd2b697443/python/GafferSceneUITest/ParameterInspectorTest.py#L553-L556

Using the `inspect()` from `Inspector` happily returns an edit to set the attribute `nothingHere`.

Regardless, moving the test to `ParameterInspector::history()` has the same effect. Fix is imminent.

Reply:
Fixed in e977afe37eef473dbb60c18ddc8a3c90dbaffe7d